### PR TITLE
Merge ascwds into cqc

### PR DESCRIPTION
**Pull request by garypolzin**
# Description
Merges ASCWDS data into CQC location data using align dates function

# How to test
Ran in branch and looks good in Athena

[Trello](https://trello.com/c/hscBnSqh/57-epic-7-merge-ascwds-workplace-clean-into-cqc-locations-clean-must)

# Developer checklist
- [X] Unit test
- [X] Linked to Trello ticket
- [X] Documentation up to date

### CODE REVIEW

File: jobs/clean_ascwds_workplace_data.py [hunk 1]
```diff
@@ -66,11 +66,13 @@ def main(source: str, destination: str):
 
     ascwds_workplace_df = create_column_with_repeated_values_removed(
         ascwds_workplace_df,
-        AWP.total_staff,
```
Comment (MarionJHolloway) [inline jobs/clean_ascwds_workplace_data.py:69]: Do we need the original columns for job calculator?

File: jobs/merge_ind_cqc_data.py [hunk 2]
```diff
@@ -3,28 +3,82 @@
...
+)
 from utils.column_names.ind_cqc_pipeline_columns import (
     PartitionKeys as Keys,
-    MergeIndCqcColumns,
-    MergeIndCqcValues,
 )
 
 PartitionKeys = [Keys.year, Keys.month, Keys.day, Keys.import_date]
 
+cqc_columns_to_import = [
```
Comment (MarionJHolloway) [inline jobs/merge_ind_cqc_data.py:21]: could we make these variable names slightly more specific to the dataset we're importing? E.g. cqc_locations_cleaned_columns_to_import? Same with the ascwds variable name below

File: jobs/merge_ind_cqc_data.py [hunk 3]
```diff
@@ -34,7 +88,45 @@ def main(
...
+        (
+            primary_df_with_secondary_import_date[secondary_import_date_column]
+            == secondary_df[secondary_import_date_column_to_drop]
+        )
+        & (
+            primary_df_with_secondary_import_date[CQCLClean.location_id]
+            == secondary_df[secondary_location_id_to_drop]
+        ),
+        how="left",
+    ).drop(secondary_import_date_column_to_drop, secondary_location_id_to_drop)
```
Comment (MarionJHolloway) [inline jobs/merge_ind_cqc_data.py:127]: You can avoid the column name changes here by referencing the originating dataframes in the drop function, e.g. 

.drop(secondary_df[secondary_import_date_column], secondary_df[location_id])

**Review by MarionJHolloway — CHANGES_REQUESTED** [review]
Only part way through this review but in meetings for a bit!

File: jobs/clean_ascwds_workplace_data.py (hunk 1 above)
Comment (garypolzin) [reply jobs/clean_ascwds_workplace_data.py:69]: this step creates a new column with duplicates removed so we still have the original columns

the '_bounded' version is the same as the original but excludes zero's

File: jobs/merge_ind_cqc_data.py (hunk 3 above)
Comment (garypolzin) [reply jobs/merge_ind_cqc_data.py:127]: yea I tried that originally but I couldn't get it to work - import date works fine with your suggestion (it's a different name) but locationId fails

The following code give the error below

```
    merged_df = primary_df_with_secondary_import_date.join(
        secondary_df,
        (
            primary_df_with_secondary_import_date[secondary_import_date_column]
            == secondary_df[secondary_import_date_column]
        )
        & (
            primary_df_with_secondary_import_date[CQCLClean.location_id]
            == secondary_df[AWPClean.location_id]
        ),
        how="left",
    ).drop(
        secondary_df[secondary_import_date_column], secondary_df[AWPClean.location_id]
    )
```

which errors with
`unit.test_merge_ind_cqc_data.MergeIndCQCDatasetTests.test_join_ascwds_data_into_merged_df failed with error: <class 'pyspark.errors.exceptions.captured.AnalysisException'> [UNRESOLVED_COLUMN.WITH_SUGGESTION] A column or function parameter with name `locationId` cannot be resolved. Did you mean one of the following? [`locationId`, `locationid`, `careHome`, `totalstaff`, `cqc_sector`].;`

Comment (garypolzin) [reply jobs/merge_ind_cqc_data.py:127]: I'll look into it a bit more

Comment (garypolzin) [reply jobs/merge_ind_cqc_data.py:127]: ah got it now - needed to rename the locationid so it's an exact match in both, and then I don't need to drop any columns

**Review by MarionJHolloway — APPROVED** [review]
Looks good to me, thanks :)

**Review by james-francis-MT — APPROVED** [review]

**Pull request merged**